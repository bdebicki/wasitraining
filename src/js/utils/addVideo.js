export default function({
	classNames,
	src,
	placeholder,
	width,
	height,
}) {
	const video = document.createElement('video');

	if (Array.isArray(classNames)) {
		classNames.forEach((className) => video.classList.add(className));
	} else {
		video.classList.add(classNames);
	}
	video.setAttribute('src', src);
	video.setAttribute('poster', placeholder);
	video.width = width;
	video.height = height;
	video.setAttribute('autoplay', '');
	video.setAttribute('loop', '');

	return video;
}
