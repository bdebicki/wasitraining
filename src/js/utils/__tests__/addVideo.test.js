import using from '../../../../testUtils/using';
import addVideo from '../addVideo';

const videoElCases = [
	{
		description: 'with one class name',
		classNames: 'bgVideo',
	},
	{
		description: 'with multiple class names',
		classNames: ['bgVideo', 'videoFullScreen'],
	},
];

describe('render video element', () => {
	using(videoElCases).describe('', ({ description, classNames }) => {
		it(description, () => {
			const videoEl = addVideo({
				classNames,
				src: './video.webm',
				placeholder: './placeholder.png',
				width: 200,
				height: 100,
			});

			expect(videoEl).toMatchSnapshot();
		});
	});
});
