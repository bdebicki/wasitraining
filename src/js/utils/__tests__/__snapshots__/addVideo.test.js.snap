// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render video element	with multiple class names 1`] = `
<video
	autoplay=""
	class="bgVideo videoFullScreen"
	height="100"
	loop=""
	poster="./placeholder.png"
	src="./video.webm"
	width="200"
/>
`;

exports[`render video element	with one class name 1`] = `
<video
	autoplay=""
	class="bgVideo"
	height="100"
	loop=""
	poster="./placeholder.png"
	src="./video.webm"
	width="200"
/>
`;
